gdk4/src/subclass/paintable.rs

```diff
@@ -0,0 +1,173 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::{Paintable, PaintableFlags, Snapshot};
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::Cast;
+
+pub trait PaintableImpl: ObjectImpl {
+    fn get_current_image(&self, paintable: &Self::Type) -> Paintable {
+        unsafe {
+            let type_ = ffi::gdk_paintable_get_type();
+            let iface = glib::gobject_ffi::g_type_default_interface_ref(type_)
+                as *mut ffi::GdkPaintableInterface;
+            assert!(!iface.is_null());
+
+            let ret = ((*iface).get_current_image.as_ref().unwrap())(
```

Comment:
The default implementation is not too useful:

```cpp
static GdkPaintable *
gdk_paintable_default_get_current_image (GdkPaintable *paintable)
{
  g_warning ("FIXME: implement by snapshotting at default size and returning a GskRendererNodePaintable");

  return paintable;
}
```

AFAIU this is a lazy way to say that this virtual method is actually mandatory, or not?

Reply:
If your paintable's content is constant & doesn't change, your return self which is the default implementation, I'm not sure why that warning's there

Reply:
Worth a GTK bug report IMHO. There should never be warnings in normal circumstances.